terraform/app.tf

```
@@ -0,0 +1,85 @@
+locals {
+  app_environment_variables = merge(try(local.infrastructure_secrets, null),
+    {
+      REDIS_URL = cloudfoundry_service_key.redis_key.credentials.uri
+    }
+  )
+  logstash_endpoint = data.azurerm_key_vault_secret.secrets["LOGSTASH-ENDPOINT"].value
+}
+
+resource "cloudfoundry_route" "apply_qts_public" {
+  domain   = data.cloudfoundry_domain.cloudapps.id
+  hostname = var.apply_qts_app_name
+  space    = data.cloudfoundry_space.space.id
+}
+
+resource "cloudfoundry_route" "apply_qts_internal" {
+  count    = local.configure_prometheus_network_policy
+  domain   = data.cloudfoundry_domain.internal.id
+  space    = data.cloudfoundry_space.space.id
+  hostname = var.apply_qts_app_name
+}
+
+resource "cloudfoundry_route" "apply_qts_education" {
+  for_each = toset(var.hostnames)
+  domain   = data.cloudfoundry_domain.education_gov_uk.id
+  space    = data.cloudfoundry_space.space.id
+  hostname = each.value
+}
+
+resource "cloudfoundry_user_provided_service" "logging" {
+  name             = var.logging_service_name
+  space            = data.cloudfoundry_space.space.id
+  syslog_drain_url = "syslog-tls://${local.logstash_endpoint}"
+}
+resource "cloudfoundry_service_instance" "postgres" {
+  name         = var.postgres_database_name
+  space        = data.cloudfoundry_space.space.id
+  service_plan = data.cloudfoundry_service.postgres.service_plans[var.postgres_database_service_plan]
+  json_params  = jsonencode(local.restore_db_backup_params)
+  timeouts {
+    create = "60m"
+    update = "60m"
+  }
+}
+
+resource "cloudfoundry_service_instance" "redis" {
+  name         = var.redis_name
+  space        = data.cloudfoundry_space.space.id
+  service_plan = data.cloudfoundry_service.redis.service_plans[var.redis_service_plan]
+}
+
+resource "cloudfoundry_service_key" "redis_key" {
+  name             = "${var.redis_name}_key"
+  service_instance = cloudfoundry_service_instance.redis.id
+}
+resource "cloudfoundry_app" "app" {
+  name                       = var.apply_qts_app_name
+  space                      = data.cloudfoundry_space.space.id
+  instances                  = var.apply_qts_instances
+  memory                     = var.apply_qts_memory
+  disk_quota                 = var.apply_qts_disk_quota
+  docker_image               = var.apply_qts_docker_image
+  strategy                   = "blue-green"
+  environment                = local.app_environment_variables
+  health_check_type          = "http"
+  health_check_http_endpoint = "/health"
```

This will be different for this app:

```suggestion
  health_check_http_endpoint = "/healthcheck"
```

We should update the other locations in the code as well :+1: